packages/alchemy/src/config/store/client.ts

```diff
@@ -0,0 +1,141 @@
+import type { PartialBy } from "viem/chains";
+import { subscribeWithSelector } from "zustand/middleware";
+import { createStore } from "zustand/vanilla";
+import type { AlchemySignerClient } from "../../signer/index.js";
+import {
+  AlchemySigner,
+  type AlchemySignerParams,
+} from "../../signer/signer.js";
+import { AlchemySignerStatus } from "../../signer/types.js";
+import { DEFAULT_IFRAME_CONTAINER_ID } from "../createConfig.js";
+import type { SupportedAccountTypes } from "../types.js";
+import type {
+  AccountState,
+  ClientState,
+  ClientStore,
+  SignerStatus,
+} from "./types.js";
+
+export type CreateClientStoreParams = {
+  client: PartialBy<
+    Exclude<AlchemySignerParams["client"], AlchemySignerClient>,
+    "iframeConfig"
+  >;
+  sessionConfig?: AlchemySignerParams["sessionConfig"];
+};
+
+export const createClientStore = (config: CreateClientStoreParams) => {
+  const clientStore =
+    typeof window === "undefined"
+      ? undefined
+      : createStore(
+          subscribeWithSelector(() => createInitialClientState(config))
+        );
+
+  addClientSideStoreListeners(clientStore);
+
+  return clientStore;
+};
+
+const createSigner = (params: CreateClientStoreParams) => {
+  const { client, sessionConfig } = params;
+  const { iframeContainerId } = client.iframeConfig ?? {
+    iframeContainerId: DEFAULT_IFRAME_CONTAINER_ID,
+  };
+
+  let iframeContainer = document.getElementById(iframeContainerId);
+  if (iframeContainer !== null) {
+    iframeContainer.innerHTML = "";
+    iframeContainer.style.display = "none";
+  } else {
+    iframeContainer = document.createElement("div");
+    iframeContainer.id = iframeContainerId;
+    iframeContainer.style.display = "none";
+    document.body.appendChild(iframeContainer);
+  }
+
+  const signer = new AlchemySigner({
+    client: {
+      ...client,
+      iframeConfig: {
+        ...client.iframeConfig,
+        iframeContainerId,
+      },
+    },
+    sessionConfig,
+  });
+
+  const search = new URLSearchParams(window.location.search);
+  if (search.has("bundle")) {
+    signer.authenticate({ type: "email", bundle: search.get("bundle")! });
+  }
```

Comment:
ultra nit: but "bundle" is a keyword and might be worth turning into a const that we can update in case we want to modify it in the future more easily if our email redirect flow changes.

not necessary to do in this pr

Reply:
oh yea good shout